pilot/pkg/model/xds_cache.go

```diff
@@ -236,7 +277,7 @@ func (l *lruCache) Add(entry XdsCacheEntry, pushReq *PushRequest, value *discove
 		return
 	}
 
-	toWrite := cacheValue{value: value, token: token}
+	toWrite := cacheValue{value: value, token: token, dependentConfigs: entry.DependentConfigs(), dependentTypes: entry.DependentTypes()}
```

Comment:
This will cause more mem cost

Reply:
it will occupy more memory than previously, yes, but i will still be much less than today, since currently the cache appends indefinitely to the indexes and never deletes from them, causing a memory leak.

Another thing is, this items are already used as keys for the `configIndex` and `typesIndex` so they are actually already kept in memory by those maps, so will we really add that much memory?

Reply:
>but i will still be much less than today, since currently the cache appends indefinitely to the indexes and never deletes from them, causing a memory leak.

I do not think so in case there are less than 60000 xds keys

Reply:
>Another thing is, this items are already used as keys for the configIndex and typesIndex so they are actually already kept in memory by those maps, so will we really add that much memory?


The struct is map[configkey]map{xds key}struct{}, it is totally different from `entry.DependentConfigs()`, so I think only little part of mem is reused. entry.DependentConfigs() returns a slice, each time it is reconstructed.